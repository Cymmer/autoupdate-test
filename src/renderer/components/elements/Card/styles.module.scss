@import '../../../scss/variables';
@import '../../../scss/themes';

.Card {
  @include theme {
    background-color: t('cardBackgroundColor');
    border-color: t('borderColor');
  }
  border-radius: 16px;
  border: 1px solid;
  text-decoration: none;

  &___clickable {
    transition: background-color 0.3s;
    cursor: pointer;

    @include theme {
      &:hover,
      &:focus,
      &:active {
        background-color: t('cardBackgroundColor--hover');
      }
    }
  }
}
