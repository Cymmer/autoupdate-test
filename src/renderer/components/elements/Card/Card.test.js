import React from 'react';
import { shallow, mount } from 'enzyme';
import { findByTestAttr } from 'codechum-app-utils';
import Card from './index';

describe('RENDERING - Card', () => {
  it('should render with default props', () => {
    const component = shallow(<Card />);
    expect(component).toMatchSnapshot();
  });

  it('should render its children', () => {
    const component = shallow(<Card>Test</Card>);
    expect(component).toMatchSnapshot();
  });
});

describe('EVENTS - Button', () => {
  it('should be clickable if it has an onClick handler', () => {
    const clickFn = jest.fn();
    const component = mount(<Card onClick={clickFn} />);

    findByTestAttr(component, 'card').simulate('click');

    expect(clickFn).toHaveBeenCalled();
  });

  it('should be clickable via ENTER key if it has an onClick handler', () => {
    const clickFn = jest.fn();
    const component = mount(<Card onClick={clickFn} />);

    findByTestAttr(component, 'card').simulate('keypress', { key: 'Enter' });

    expect(clickFn).toHaveBeenCalled();
  });

  it('should be clickable via SPACE key if it has an onClick handler', () => {
    const clickFn = jest.fn();
    const component = mount(<Card onClick={clickFn} />);

    findByTestAttr(component, 'card').simulate('keypress', { key: ' ' });

    expect(clickFn).toHaveBeenCalled();
  });
});
