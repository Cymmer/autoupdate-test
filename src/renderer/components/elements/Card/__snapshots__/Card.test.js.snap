// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Card should render its children 1`] = `
<div
  className="Card"
  data-test="card"
  id={null}
  onClick={null}
  onKeyPress={null}
  onMouseEnter={null}
  onMouseLeave={null}
  role={null}
  tabIndex={null}
>
  Test
</div>
`;

exports[`RENDERING - Card should render with default props 1`] = `
<div
  className="Card"
  data-test="card"
  id={null}
  onClick={null}
  onKeyPress={null}
  onMouseEnter={null}
  onMouseLeave={null}
  role={null}
  tabIndex={null}
/>
`;
